libs/hbb_common/src/socket_client.rs

```diff
@@ -154,7 +154,7 @@ pub fn ipv4_to_ipv6(addr: String, ipv4: bool) -> String {
     addr
 }
 
-async fn test_target(target: &str) -> ResultType<SocketAddr> {
+pub async fn test_target(target: &str) -> ResultType<SocketAddr> {
```

Comment:
socks5 proxy tested?

Reply:
The ability to establish a NATS MQ connection via a user-specified socks5 proxy has not yet been implemented. This is mainly because [nats.rs](https://github.com/nats-io/nats.rs) doesn't expose the necessary API. Instead, it just using `TcpStream::connect` directly ([async-nats/src/connector.rs](https://github.com/nats-io/nats.rs/blob/c661d2147ccbbfc25a32fad8e5b9c1cb9ca077df/async-nats/src/connector.rs#L291C27-L291C27)). We might need to fork our own version to replace it with other `impl AsyncWrite + AsyncRead + Send + Unpin`, e.g, getting from `FramedStream`.

Reply:
We have to fork then.